packages/quantic/force-app/main/default/lwc/quanticModal/__tests__/quanticModal.test.js

```diff
@@ -0,0 +1,193 @@
+};
+
+describe('c-quantic-modal', () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  describe('when using default options', () => {
+    it('should not display the modal by default', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+      const modal = element.shadowRoot.querySelector(selectors.modal);
+      expect(modal.classList.contains('modal_hidden')).toBe(true);
+    });
+
+    it('should have slide-to-top animation class by default', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const modal = element.shadowRoot.querySelector(selectors.modal);
+      expect(modal.classList.contains('hidden-modal_slide-to-top')).toBe(true);
+    });
+
+    it('should display the modal when openModal is called', async () => {
+      const element = createTestComponent();
+      element.openModal();
+      await flushPromises();
+
+      const modal = element.shadowRoot.querySelector(selectors.modal);
+      expect(modal.classList.contains('modal_hidden')).toBe(false);
+
+      expect(
+        element.shadowRoot.querySelector(selectors.modalHeader)
+      ).not.toBeNull();
+      expect(
+        element.shadowRoot.querySelector(selectors.modalContent)
+      ).not.toBeNull();
+      expect(
+        element.shadowRoot.querySelector(selectors.modalFooter)
+      ).not.toBeNull();
+
+      expect(modal.classList.contains('full-screen')).toBe(false);
+      expect(modal.classList.contains('part-screen')).toBe(true);
+    });
+
+    it('should hide the modal when closeModal is called', async () => {
+      const element = createTestComponent();
+      element.openModal();
+      await flushPromises();
+      element.closeModal();
+      await flushPromises();
+      const modal = element.shadowRoot.querySelector(selectors.modal);
+      expect(modal.classList.contains('modal_hidden')).toBe(true);
+    });
+
+    it('should focus on first focusable element when modal opens', async () => {
+      const mockFirstFocusableElement = {focus: jest.fn()};
+      // @ts-ignore
+      getFirstFocusableElement.mockImplementation(
+        () => mockFirstFocusableElement
+      );
+
+      const element = createTestComponent();
+      await flushPromises();
+
+      const headerSlot = element.shadowRoot.querySelector('slot[name=header]');
+      const contentSlot =
+        element.shadowRoot.querySelector('slot[name=content]');
+      const footerSlot = element.shadowRoot.querySelector('slot[name=footer]');
+
+      const headerSlotSpy = jest
+        .spyOn(headerSlot, 'assignedElements')
+        .mockImplementation(() => [document.createElement('div')]);
+      const contentSlotSpy = jest
+        .spyOn(contentSlot, 'assignedElements')
+        .mockImplementation(() => [document.createElement('div')]);
+      const footerSlotSpy = jest
+        .spyOn(footerSlot, 'assignedElements')
+        .mockImplementation(() => [document.createElement('div')]);
```

Comment:
What's the point of spying on all three of these slots especially that we are not doing much with them,
we can spy on just one of them and then assure  that `getFirstFocusableElement ` has been called.